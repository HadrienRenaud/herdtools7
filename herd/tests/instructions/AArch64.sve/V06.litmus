AArch64 V06
(* Contiguous load three-word structures to three vectors (scalar index) *)
{
 int x[12] = {1,2,3,1,2,3,1,2,3,1,2,3};

 0:X0=x;
}
P0                                         ;
 MOV X2,#0                                 ;
 PTRUE P0.S,VL4                            ;
 LD3W {Z0.S,Z1.S,Z2.S},P0/Z,[X0,X2,LSL #2] ;

forall(0:V0.4S = {1,1,1,1} /\ 0:V1.4S = {2,2,2,2} /\ 0:V2.4S = {3,3,3,3})
