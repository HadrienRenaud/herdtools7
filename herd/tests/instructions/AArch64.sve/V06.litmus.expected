Test V06 Required
States 1
0:V0.4S={1,1,1,1}; 0:V1.4S={2,2,2,2}; 0:V2.4S={3,3,3,3};
Ok
Witnesses
Positive: 1 Negative: 0
Flag Scalable-Vector-Extensions-is-work-in-progress
Condition forall (0:V0.4S={1,1,1,1} /\ 0:V1.4S={2,2,2,2} /\ 0:V2.4S={3,3,3,3})
Observation V06 Always 1 0
Hash=0b0d64e1c8f343cdce7976205bc08f2b

